### ISSUE TITLE
Implement Multi-Task deep learning with Tensorflow

### ISSUE THREAD

**Issue by MammadTavakoli** [pkmital/tensorflow_tutorials#17]
How can implement Multi-Task deep learning with Tensorflow? how can i found any sample? 

![Multi-Task deep learning](https://camo.githubusercontent.com/9b5b091d632c5b923b706016ca15ee56e70c9958/687474703a2f2f322e62702e626c6f6773706f742e636f6d2f2d74736736742d63547857302f56626e31416c51776e72492f414141414141414143546b2f487a5548794e62547478492f73313630302f53637265656e25324253686f74253242323031352d30372d33302532426174253242253235453125323538342532353842253235453125323538352532354139253235453125323538342532353932253235453125323538352532354145253242362e35352e30312e706e67)

**Comment by pkmital** [comment]
You could create multiple linear layers from the same input, e.g.:

```
from libs.connections import linear
x = tf.placeholder(tf.float32)
attr1 = linear(x, 32)
attr2 = linear(x, 32)
...
```

then your loss function should incorporate each of these outputs somehow, or you could train them based on multiple loss functions.

**Comment by MammadTavakoli** [comment]
thanks for your replay.

I don't understand how train the parameters in BP. Are they learned in the same time? 
How multiple attribute can influenced from same parameters?

Where I can find a simple sample?

**Comment by pkmital** [comment]
You could create an optimizer given loss functions combining some loss from each attribute, e.g. summing all of them, or create multiple optimizers and train them in a stochastic manner, e.g. randomly pick a subset of them to update.
